file_stream/file_stream.go

```diff
@@ -0,0 +1,111 @@
+package file_stream
+
+import (
+	"context"
+	"io"
+	"log"
+	"os"
+
+	"github.com/fsnotify/fsnotify"
+)
+
+type FileStream struct {
+	file_name string
+	reader    *os.File
+	watcher   *fsnotify.Watcher
+
+	// Used for get command completion notifications
+	Done chan bool
```

Comment:
So to close a `FileStream` do I call `Close` or send something via the `Done` channel?

Reply:
My thinking was that I'd use this later to communicate the death of the underlying process to all readers to let them know. But now I don't think I actually need it. Removed that part.